### DISCUSSION TITLE
Name change discussion

### DISCUSSION THREAD

**Discussion by iameskild** [nebari-dev/nebari#964] [General]
name change discussion

**Comment by iameskild** [comment]
We spoke briefly during one recent meeting about not only coming up with a name but perhaps a **_theme_** for this new organization. I suggested a tree/forest theme. 

Copying over my recent slack message on this topic:
> Qhub offers an open platform that is highly customizable and extensible (especially if more of our ideas become a reality) that enables individuals, businesses and researches to build and develop ideas and products of their own. Trees create an ecosystem (forests) that enables other plants and animals to live and thrive. With that, here are just a few ideas:
> - heartwood
> - willow
> - banyan
> - arboretum

**Comment by tylerpotts** [comment]
`YadaHub`

Yet Another Datascience Hub

**Comment by viniciusdc** [comment]
`Collatz`, as the Collatz conjecture, Qhub is a work in progress project. From multiple different places, clients, or external users we always aim for a stable point well maintained and easier to work with, always targeting that final 1. (kind of)

**Reply by iameskild** [reply]
This is great! 
I just saw a youtube video on this last week 🤯

**Comment by rsignell-usgs** [comment]
I love the tree idea!  How about "aspen"?

![2021-12-14_14-35-03](https://user-images.githubusercontent.com/1872600/146067213-8b1f4ad7-6d25-4139-851d-c76e66bd8be9.png)


It's on the [fastest growing trees list](https://arbordayblog.org/landscapedesign/the-fastest-fast-growing-trees/) from arborday

And from [fed.us](https://www.fs.fed.us/wildflowers/beauty/aspen/grow.shtml) it's described as:

> "... quick to pioneer disturbed sites"

> " Aspen is noted for its ability to regenerate vegetatively by shoots and suckers arising along its long lateral roots. Root sprouting results in many genetically identical trees, in aggregate called a "clone". All the trees in a clone have identical characteristics and share a root structure "

> "aspen is America's liveliest tree. With just the slightest breeze, its round leaves tremble almost incessantly, like thousands of fluttering butterfly wings."

**Reply by iameskild** [reply]
Beautiful!

**Reply by dharhas** [reply]
So I like aspen but there is already a python package called aspen (aspen.io). Tt is different enough that we may still be able to use the name but we wouldn't be able to have a python package with that name.

**Reply by rsignell-usgs** [reply]
Grrr. I should have checked that first.  I'm checking to see if we might be able to get the name "populus" which is the latin name for aspen. That's taken too, but the package is no longer used (the functionality has moved to a new package) so I'm checking with the folks who used to develop it.  Would be a pretty awesome name, even better than aspen!

**Comment by magsol** [comment]
Suggested by @trallard : Avalon

**Reply by magsol** [reply]
Also: Arcadia

**Comment by tylerpotts** [comment]
Mycorr

Short for [Mycorrhizal network](https://en.wikipedia.org/wiki/Mycorrhizal_network)

Still fits within the tree/forest theme

**Reply by magsol** [reply]
Maybe MycorrHub? Admittedly kind of a play on swapping "hub" in for "network", but maybe a little too loaded.

**Comment by costrouc** [comment]
Ideas: Alder

**Reply by magsol** [reply]
Suggested by @tonyfast : Qalder

**Reply by trallard** [reply]
I think we should avoid the Q in the name

**Comment by rsignell-usgs** [comment]
"alder" is already on pypi: https://pypi.org/project/alder/
nothing since 2015, however, so perhaps the maintainer https://pypi.org/user/Samuel.Smith would be willing to give it up.

"qalder" is probably out since dropping the "q" was the primary reason to do the name change in the first place, right?

**Reply by magsol** [reply]
That's my inclination. Even though never explicitly stated anywhere, the implicit and widespread assumption is that the "Q" references Quansight; therefore including it in any future name would kind of defeat the purpose.

(even though I still maintain it stands for "Quinn" :) )

**Reply by magsol** [reply]
@rsignell-usgs I can reach out to the alder creator. Do you have a template you used with populus?

**Reply by rsignell-usgs** [reply]
You can raise an issue with pypi for a transfer of name, but I'd do that as a last resort, as apparently they rarely grant these requests. 

The first thing to try is to contact the Maintainer (Samuel Smith) and see if he is willing to give it up.  If he is, he can add you as a maintainer, and then you can remove him, delete the project, and then immediately create it again on pypi

**Comment by rsignell-usgs** [comment]
I am still pursuing whether we might be able to use "populus"  

I asked a few users and they said they thought "hub" should remain in the name. 
It looks like we could use "hubby", if that's of interest:
https://github.com/thusoy/blag/issues/27#issuecomment-1007356019

**Reply by rsignell-usgs** [reply]
So @magsol , what do you think about "hubby"?  (I kind of like it)

**Reply by magsol** [reply]
Hmm. I'm kind of indifferent (the knee-jerk context I have is the pet name for "husband" which is an odd juxtaposition), but I'm totally fine with it if everyone else likes it.

**Reply by trallard** [reply]
All I could think of was what @magsol  mentioned

**Comment by magsol** [comment]
@tonyfast : Warren

**Comment by iameskild** [comment]
`EntHub`

`Ent` 
- tree people of Lord of the Rings 
- Enterprise or professional, this is JupyterHub plus more

**Comment by magsol** [comment]
@iameskild : hazel

**Comment by iameskild** [comment]
- WillowCore
- Salix 
(genus which Willows belong to)

**Reply by dharhas** [reply]
How about `spindle` - a rod or pin serving as an axis that revolves or on which something revolves.

**Reply by tylerpotts** [reply]
`spindlehub` is a unique name when Google searching

**Reply by tylerpotts** [reply]
Also according to Google on "Spindle vs Hub": `The spindle carries the hub...`

**Comment by cgarciae** [comment]
Don't mean to add to the confusion but how about a Jupiter-related name?

https://www.google.com/search?q=jupiter+asteroides

**Comment by rsignell-usgs** [comment]
Several I've polled in the community think "hub" should be in the name. 
How about "Hub9" (a play on Cloud9)

**Comment by magsol** [comment]
Wanted to provide an update on the polling from the Slack chat.

Polling frontrunners:
 - Populus
 - Salix
 - Alder
 - Spindle

Honorable mention:
 - Aspen (Travis really likes it, so here it is)

Latest entries:
 - Hub9
 - [Nebari](https://en.wikipedia.org/wiki/Bonsai_aesthetics#Nebari)

Outstanding issues:
 - If we went with "Populus", "Aspen", or "Alder" we'd need to acquire the PyPI projects first
 - The presence or absence of the "-Hub" suffix changes preferences for core name

**Comment by tupui** [comment]
I have something which is not a tree species, but a general feature of a Bonsai.

**Nebari**: it's the base of a Bonsai, the main roots at the foundation of the tree. When the roots are homogeneous and large, we talk about having a strong Nebari. I believe the symbol could fit with the product :smiley:

**Reply by iameskild** [reply]
I think this is a wonderful name! Short, unique and great symbolism :)

**Reply by trallard** [reply]
My vote goes to Nebari and even if we want to add the Hub it has a nice ring to it "Nebari-Hub"

**Reply by dharhas** [reply]
'nebari' wins in my opinion, both from a unique, nice sounding name and from having a great meaning.

**Reply by rsignell-usgs** [reply]
Sounds good to me!

**Reply by rsignell-usgs** [reply]
Having slept on it, I like the root system idea, and anything with 3 syllables and ends with an "i" sounds cool, but I'm not sure bonsai is best to identify with.   Just to play devil's advocate:

Wikipedia says: "Purposes of bonsai are primarily contemplation for the viewer, and the pleasant exercise of effort and ingenuity for the grower".   

Also kind of the opposite of fast growing!

Snarky subtitles for the project if we choose this name?   
-- "Wee is us"  
-- "Go small or go home!"

Just wanted to pull it out to see if it snaps back....

**Reply by tupui** [reply]
Disclaimer: I love Bonsai 😍 I got a few when I was in France and sadly had to give them away when I left to Austria.

These are valid points!

Some elements to defend it 😉

I think we can easily frame a Bonsai to our advantage still. Bonsai is really an art and lifestyle for some people. Growing a Bonsai requires knowledge, patience, time and care. It's a long term relationship you need to have with your tree. During its long life, you will add features (pruning, orientation, dead wood, change branch orientations, etc.) and oversee its grow on a daily basis. All this to have a strong and healthy tree. I think this is what we can wish for a product, API, etc. Grow, yes, but with planning/vision/users in mind which all take time to reflect upon. Then you will have a strong product which corresponds to something, someone.

To the size part, I don't think/hope we would get critics on that in 2022 😅 But again, you can also see this as a cornerstone. A small gem packed with all the features you could dream of. A real tree with all its features, in a small, light, human size, understandable packaging. Just a nitpick about the signification of Bonsai, it translate to _tree in a pot_. There are actually [specimens](https://www.flickr.com/photos/7940016@N04/2512888082/in/photolist-4Q4d5A-4Ti8CU-4YhuJn-5dbgbs-5nLhkr-5Ck9bB-5Mgfj2-5NC7tg-5VyYk1-6217Mb-65gHAh-65gJfy-68otpj-68otpo-69nrLW-69ntwm-69NMgk-6cXZhX-6d1wqV-6dcsov-6nkCxd-6oxbuD-6qvLkz-6qvNDV-6qzMFs-6qzNSG-6qzRob-6qzVaw-6qzY7u-6qzYSb-6JdJNK-6JhRr5-6Mbquu-6Ru1bi-6ToJaM-6ToJb4-6TTNN6-6UiHjY-6X7DC4-75ZDyT-779bFx-7bp9ML-7hvgrp-7hz5TW-7j4CBe-7j8w2f-7kZEQS-7n1yKX-7sSNpB-7vtbyy-cDyozs) a few meters tall.

**Reply by rsignell-usgs** [reply]
@tupui , I'm overwhelmed by your passion!    Green light from me!

**Reply by iameskild** [reply]
🌲 🌲 🌲

**Reply by teoliphant** [reply]
nebari itself is a hard name to secure for websites and github -- but nebarihub is available.

**Reply by tupui** [reply]
Still available on pypi as well.

**Reply by magsol** [reply]
@trallard's suggestion of `nebari-dev` as the GH org feels like the best route, IMO. We can still get `nebari` on pypi, and figure out what base URL we want, but in particular her point about a potential future `nebari-hub` would be much easier to include using that convention.

There is an existing nebari org on GitHub (appears unused since 2015, at least publicly) and a nebari.org domain (appears unused since 2011, at least publicly), as well as an actively-maintained Rust project called nebari. That's the extent I could find.

**Reply by magsol** [reply]
nvm, `nebari-dev` appears to be taken on GH as well, though it shows no public members, repos, or activity of any kind.

**EDIT** lol claimed by Travis, we're good :)

**Reply by tupui** [reply]
> **EDIT** lol claimed by Travis, we're good :)

Is this decided yet then?? If we start reserving things, let me know if I can help (can do PyPi for instance). EDIT: done

**Comment by magsol** [comment]
Yes! Travis got nebari-dev on github and registered the nebari.dev TLD so
go right ahead and nab the nepari pypi name.

On Tue, Jan 25, 2022 at 18:03 Pamphile Roy ***@***.***> wrote:

> *EDIT* lol claimed by Travis, we're good :)
>
> Is this decided yet then?? If we start reserving things, let me know if I
> can help (can do PyPi for instance)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Quansight/qhub/discussions/964#discussioncomment-2046925>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABBB6LXA77YXSBNPRXXL5DUX4T35ANCNFSM5JRZNZXA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
iPhone’d

**Reply by tupui** [reply]
Done ✅ https://pypi.org/project/nebari/ I will put the empty repo on our GH tomorrow and discuss next steps (who to add etc.).

**Reply by rsignell-usgs** [reply]
I mentioned at the Pangeo weekly call that qhub was going to rebrand to nebari and they said "won't that be confusing with Napari?  https://napari.org/  

Not really, right?

**Comment by magsol** [comment]
If folks can keep Quantstack and Quansight straight then Nebari and Napari
shouldn’t be too difficult.

On Fri, Feb 18, 2022 at 09:36 Rich Signell ***@***.***> wrote:

> I mentioned at the Pangeo weekly call that qhub was going to rebrand to
> nebari and they said "won't that be confusing with Napari?
> https://napari.org/
>
> Not really, right?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Quansight/qhub/discussions/964#discussioncomment-2205167>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABBB6MKZLBYEHEWQOJYSYTU3ZKNFANCNFSM5JRZNZXA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
iPhone’d

**Comment by trallard** [comment]
🤷🏽‍♀️  it should be ok tbh - also if we are really worried about the confusion we can always refer to it as Nebari-Hub
